content/300-guides/050-database/900-turso.mdx

```diff
@@ -0,0 +1,183 @@
+
+## Connect and query your primary database
+
+1. Create a database on Turso if you don't have one already:
+
+   ```sh
+   turso db create turso-prisma-db
+   ```
+
+2. Retrieve the database's connection string:
+
+   ```sh
+   turso db show turso-prisma-db
+   ```
+
+3. Create an authentication token that will allow you to connect to the database:
+
+   ```sh
+   turso db tokens create turso-prisma-db
+   ```
+
+4. Update your `.env` file with the authentication token and connection string:
+
+   ```text
+   TURSO_AUTH_TOKEN="eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9..."
+   TURSO_DATABASE_URL="libsql://turso-prisma-db-user.turso.io"
+   ```
+
+5. Enable the `driverAdapters` Preview feature flag:
+
+   ```prisma highlight=3;add
+   generator client {
+     provider        = "prisma-client-js"
+     previewFeatures = ["driverAdapters"]
+   }
+   
+   datasource db {
+     provider = "sqlite"
+     url      = "file:./dev.db"
+   }
+   ```
+
+6. Generate Prisma Client:
+
+   ```sh
+   npx prisma generate
+   ```
+
+7. Install the libSQL database client and Prisma driver adapter for libSQL packages:
+
+   ```sh
+   npm install @libsql/client @prisma/adapter-libsql
+   ```
+
+8. Update your Prisma Client instance with the following snippet:
+
+   ```ts
+   import { PrismaClient } from '@prisma/client'
+   import { PrismaLibSQL } from '@prisma/adapter-libsql'
+   import { createClient } from '@libsql/client'
+
+   const libsql = createClient({
+     url: `${process.env.TURSO_DATABASE_URL}`,
+     authToken: `${process.env.TURSO_AUTH_TOKEN}`,
+   })
+
+   const adapter = new PrismaLibSQL(libsql)
+   const prisma = new PrismaClient({ adapter })
+   ```
+
+You can use Prisma Client as you normally would with full type-safety in your project.
+
+## How to manage schema changes
+
+Prisma Migrate and Introspection workflows are currently not supported when working with Turso. This is because Turso uses HTTP to connect to your database, which Prisma Migrate doesn't support.
+
+To update your database schema:
+
+1. Generate a migration file using `prisma migrate dev` against a local SQLite database:
+
+```sh
+npx prisma migrate dev --name init # Migration name
+```
+
+2. Apply the migration using Turso's CLI:
+
+```sh
+turso db shell turso-prisma-db < ./prisma/migrations/20230922132717_init/migration.sql # Replace `20230922132717_init` with the existing migration
+```
+
+This workflow does not support tracking the history of applied migrations.
+
+## How to connect and query from an embedded replica
+
+Turso supports [embedded replicas](https://blog.turso.tech/introducing-embedded-replicas-deploy-turso-anywhere-2085aa0dc242). Turso's embedded replicas enable you to have a copy of your primary, remote database _inside_ your application. This is similar to having a local SQLite database.
```

Comment:
> This is similar to having a local SQLite database.

I don't think it is. A local SQLite database is not synced from a remote server or kept up to date with remote changes.
You probably want to say that this then behaves like a local SQLite file in that queries are super fast and similar.